src/nat/utils/decorators.py

```
@@ -0,0 +1,198 @@
+        removal_version: The version when the function will be removed
+        replacement: What to use instead of this function
+        reason: Why the function is being deprecated
+        feature_name: Optional name of the feature that is deprecated
+        metadata: Optional dictionary of metadata to log with the warning
+    """
+    if function_name not in _warning_issued:
+        # Build the deprecation message
+        if feature_name:
+            warning_message = f"The {feature_name} feature is deprecated"
+        else:
+            warning_message = f"This function is deprecated"
```

Ruff (0.13.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)